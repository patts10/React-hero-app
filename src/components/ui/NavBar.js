import React, { useContext } from "react";
import { Link, NavLink, useNavigate } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { types } from "../../types/types";

export const Navbar = () => {

  const navigate = useNavigate();
  const { user, dispatch } = useContext(AuthContext);

  const handleLogout = () => {

    dispatch({ type: types.logout });
    
    navigate('/login', {
      replace: true
    })
  };

  return (
    <nav className="relative w-full flex flex-wrap items-center justify-between py-3 bg-gray-900 text-gray-200 shadow-lg navbar navbar-expand-lg navbar-light">
      <div className="container-fluid w-full flex flex-wrap items-center justify-between px-6">
        <div className="collapse navbar-collapse flex flex-row items-center w-full">
          <Link className="text-xl text-white pr-2 font-semibold" to="/">
            Asociaciones
          </Link>
          <ul className="navbar-nav flex flex-row pl-0 list-style-none mr-auto w-full">
            <li className="nav-item p-2">
              <NavLink
                className={({ isActive }) =>
                  "nav-item " +
                  (isActive
                    ? "active"
                    : "opacity-60 hover:opacity-90 focus:opacity-80 p-0")
                }
                to="/marvel"
              >
                Marvel
              </NavLink>
            </li>
            <li className="nav-item p-2">
              <NavLink
                className={({ isActive }) =>
                  "nav-item " +
                  (isActive
                    ? "active"
                    : "opacity-60 hover:opacity-90 focus:opacity-80 p-0")
                }
                to="/dc"
              >
                Dc
              </NavLink>
            </li>
            <li className="nav-item p-2">
              <NavLink
                className={({ isActive }) =>
                  "nav-item " +
                  (isActive
                    ? "active"
                    : "opacity-60 hover:opacity-90 focus:opacity-80 p-0")
                }
                to="/search"
              >
                Search
              </NavLink>
            </li>
            <li className="nav-item p-2 absolute right-0 mx-5">
              <span className="text-sky-600 mx-3 text-info">{ user.name }</span>
              <button
                className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
                onClick={ handleLogout }
              >
                Logout
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
