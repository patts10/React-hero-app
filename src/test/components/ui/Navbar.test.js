import { mount } from "enzyme";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import { Navbar } from "../../../components/ui/NavBar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Tests on <Navbar />", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      name: "Patts",
      logged: true
    }
  };
  const wrapper = mount(
    <AuthContext.Provider value={ contextValue }>
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route  path='/' element={<Navbar />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test("should show correctly", () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Patts");
  });

  test("should call logout, call navigate and dispatch with args", () => {

    wrapper.find("button").simulate("click");
    expect( contextValue.dispatch ).toHaveBeenCalled();
    expect( contextValue.dispatch ).toHaveBeenCalledWith({'type': types.logout});
    expect( mockNavigate ).toHaveBeenCalled();
    expect( mockNavigate ).toHaveBeenCalledWith("/login", {"replace": true});
    // expect('handleLogout').toHaveBeenCalled();
  });
});
