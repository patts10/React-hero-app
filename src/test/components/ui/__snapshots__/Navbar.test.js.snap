// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Navbar /> should show correctly 1`] = `
<nav
  className="relative w-full flex flex-wrap items-center justify-between py-3 bg-gray-900 text-gray-200 shadow-lg navbar navbar-expand-lg navbar-light"
>
  <div
    className="container-fluid w-full flex flex-wrap items-center justify-between px-6"
  >
    <div
      className="collapse navbar-collapse flex flex-row items-center w-full"
    >
      <a
        className="text-xl text-white pr-2 font-semibold"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <ul
        className="navbar-nav flex flex-row pl-0 list-style-none mr-auto w-full"
      >
        <li
          className="nav-item p-2"
        >
          <a
            className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
        </li>
        <li
          className="nav-item p-2"
        >
          <a
            className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
            href="/dc"
            onClick={[Function]}
          >
            Dc
          </a>
        </li>
        <li
          className="nav-item p-2"
        >
          <a
            className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="nav-item p-2 absolute right-0 mx-5"
        >
          <span
            className="text-sky-600 mx-3 text-info"
          >
            Patts
          </span>
          <button
            className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
